File: app/src/main/java/com/frankly/randomdog/ui/doggenerator/DogGeneratorViewModel.java
package com.frankly.randomdog.ui.doggenerator;

import androidx.databinding.ObservableField;
import androidx.lifecycle.ViewModel;

public class DogGeneratorViewModel extends ViewModel {
    // TODO: Implement the ViewModel

    public DogGeneratorViewModel() {}

    public ObservableField<String> imageURL = new ObservableField<>();

    public ObservableField<String> getImageURL() {
        return imageURL;
    }

    public void setImageURL(ObservableField<String> imageURL) {
        this.imageURL = imageURL;
    }
}

File: app/src/main/java/com/randomdog/ui/doggenerator/viewmodel/DogGeneratorViewModel.kt
package com.randomdog.ui.doggenerator.viewmodel

import androidx.databinding.ObservableField
import androidx.lifecycle.ViewModel

/**
 * @author: ArshVardhanAtreya <arshvardhan></arshvardhan>@yahoo.co.in>
 */
class DogGeneratorViewModel : ViewModel() {
    private var imageURL: ObservableField<String?>? = ObservableField()
    fun getImageURL(): ObservableField<String?>? {
        return imageURL
    }

    fun setImageURL(imageURL: ObservableField<String?>?) {
        this.imageURL = imageURL
    }
}